tests/App/config/services.yml

```diff
@@ -42,10 +42,14 @@ services:
         class: Sonata\AdminBundle\Tests\App\Datagrid\ProxyQuery
 
     Sonata\AdminBundle\Tests\App\Admin\FooAdmin:
-        arguments: [~, Sonata\AdminBundle\Tests\App\Model\Foo, Sonata\AdminBundle\Controller\CRUDController]
+        arguments: [~, Sonata\AdminBundle\Tests\App\Model\Foo, 'sonata.admin.controller.crud']
```

Comment:
Do we have a way to make this working with `Sonata\AdminBundle\Controller\CRUDController` to avoid the BC-break ?

Does an alias could work ?

Reply:
I guess so, what we can do is to create `sonata.admin.controller.crud` in `3.x` and deprecate `Sonata\AdminBundle\Controller\CRUDController`, but I don't know why someone would use `Sonata\AdminBundle\Controller\CRUDController`.

Reply:
> but I don't know why someone would use `Sonata\AdminBundle\Controller\CRUDController`.

I find more natural to pass a class rather a name.
And some people can prefer to pass the default values.

Reply:
And why https://github.com/sonata-project/SonataAdminBundle/pull/6793/commits/11217d12bfabb651dfec8863a94a46442d93ccd2 is not working ?

Reply:
If we don't support the old way, we need to update
https://github.com/sonata-project/SonataAdminBundle/blob/master/docs/cookbook/recipe_knp_menu.rst#hiding-menu-items
https://github.com/sonata-project/SonataAdminBundle/blob/master/docs/cookbook/recipe_knp_menu.rst#show-menu-item-without-treeview

And I don't know if we need to change the explanation somewhere else.

Reply:
> And why [11217d1](https://github.com/sonata-project/SonataAdminBundle/commit/11217d12bfabb651dfec8863a94a46442d93ccd2) is not working ?

maybe the alias needs to be public then?

Reply:
Shall we include the "fix" to make the alias public here?

Reply:
We changed in `4.0` the services to not use FQCN (at least some of them I think), IMHO we shouldn't promote using it, maybe triggering a deprecation in `3.x` if the user has specified `CRUDController::class` explicitly as a third parameter.

But I'll add it if you think it would be useful.

Reply:
Yeah either triggering a deprecation or making the alias public. Both seem fine to me :blush:

Reply:
My sugestion is remove controller from `__constructor()`. This functionality can be replace by improving `RouteCollecton` to allow do something like this:
```php
final class MediaAdmin extends AbstractAdmin
{
    protected function configureRoutes(RouteCollection $collection)
    {
        $collection->add('myCustom'); // Action gets added automatically
        $collection->add('view', $this->getRouterIdParameter().'/view');

        // new options
        $collection->add('App/Controller/CustomCRUDController::myCustom'); // Action gets added automatically
        $collection->add('App/Controller/CustomCRUDController::view', $this->getRouterIdParameter().'/view');
    }
}
```

In this case we can also split `CRUDController` to `*Action` which extends `BaseCRUDAction`.

Reply:
I'd rather remove it from the `__construct` too, but not sure how. 

Let's use https://github.com/sonata-project/SonataAdminBundle/pull/6796 for this.

Reply:
Can we move it to the `sonata_admin.admin` tag? It will be more obvious. After study this:
https://github.com/sonata-project/SonataAdminBundle/blob/a17000609e5b1d64c0102606a06267f75b10f6b5/src/Route/RouteCollection.php#L70-L105
We can do it and remove `RouteCollection::baseController` property:
```php
final class MediaAdmin extends AbstractAdmin
{
    protected function configureRoutes(RouteCollection $collection)
    {
        $collection->add('myCustom'); // Action gets added automatically
        $collection->add('view', $this->getRouterIdParameter().'/view');

        // new options
        $collection->add('myCustom', null, ['_controller' => 'App/Controller/CustomCRUDController']);
        $collection->add('myCustom', null, ['_controller' => 'app.controller.custom_contoller']);
    }
}
```

Reply:
All of the 3 parameters of an admin: code, class and controller could (and should IMHO) at least be moved to the `sonata_admin.admin` instead.
I have already created this issue: https://github.com/sonata-project/SonataAdminBundle/issues/6723